applications/asset_tracker_v2/src/modules/atv2_led_module.c

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <logging/log.h>
+#include "atv2_led_module.h"
+#include <device.h>
+#include <caf/events/led_event.h>
+#include "events/led_state_event.h"
+#include <caf/events/module_state_event.h>
+
+#if (IS_ENABLED(CONFIG_LED_USE_PWM))
+#include "led_state_def_thingy91.h"
+#else
+#include "led_state_def_nrf9160dk.h"
+#endif
+
+LOG_MODULE_REGISTER(led, CONFIG_LED_CONTROL_LOG_LEVEL);
+
+
+static void send_led_event(size_t led_id, const struct led_effect *led_effect)
+{
+	__ASSERT_NO_MSG(led_effect);
+	__ASSERT_NO_MSG(led_id < LED_ID_COUNT);
+
+	struct led_event *event = new_led_event();
+
+	event->led_id = led_id;
+	event->led_effect = led_effect;
+	EVENT_SUBMIT(event);
+}
+
+static void update_pwm_led(enum led_state state)
+{
+	send_led_event(led_map[LED_1],   &asset_tracker_led_effect[LED_TURN_OFF]);
+	switch (state)	{
+	case LED_LTE_CONNECTING:
+		send_led_event(led_map[LED_1],  &asset_tracker_led_effect[LED_LTE_CONNECTING]);
+		break;
+	case LED_GPS_SEARCHING:
+		send_led_event(led_map[LED_1],  &asset_tracker_led_effect[LED_GPS_SEARCHING]);
+		break;
+	case LED_CLOUD_PUBLISHING:
+		send_led_event(led_map[LED_1],  &asset_tracker_led_effect[LED_CLOUD_PUBLISHING]);
+		break;
+	case LED_ACTIVE_MODE:
+		send_led_event(led_map[LED_1],  &asset_tracker_led_effect[LED_ACTIVE_MODE]);
+		break;
+	case LED_PASSIVE_MODE:
+		send_led_event(led_map[LED_1],  &asset_tracker_led_effect[LED_PASSIVE_MODE]);
+		break;
+	case LED_ERROR_SYSTEM_FAULT:
+		send_led_event(led_map[LED_1],  &asset_tracker_led_effect[LED_ERROR_SYSTEM_FAULT]);
+		break;
+	case LED_FOTA_UPDATE_REBOOT:
+		send_led_event(led_map[LED_1],  &asset_tracker_led_effect[LED_FOTA_UPDATE_REBOOT]);
+		break;
+	default:
+		LOG_WRN("Unrecognized LED state event send");
+		break;
+	}
+
+
+}
+
+static void update_gpio_led(enum led_state state)
+{
+	send_led_event(led_map[LED_1],   &asset_tracker_led_effect[LED_TURN_OFF]);
```

Comment:
we don't need to send two events for one led

Also note that we actually don't need update_pwm_led vs update_gpio_led. You only need information which led is used by states CONNECTING, SEARCHING, PUBLISHING, ACTIVE and PASSIVE. This could be provided in _def header. In any of these states remaining LEDs are off so you need to know led count.
Fault and reboot should apply the same effect to all leds.